### PR TITLE
add all completed k8 tenets

### PR DESCRIPTION
**Pull request by mullikine**
completed k8 tenets. tested or otherwise.

### CODE REVIEW

File: tenets/codelingo/k8/use-informer-not-list/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+tenets:
```
Comment (rstorr) [inline tenets/codelingo/k8/use-informer-not-list/codelingo.yaml]: This tenet just finds funcs with bool params ?

File: tenets/codelingo/k8/well-named-lock/codelingo.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      codelingo/docs:
+        title: Well Named Lock
+        body: |
+          A lock should be called `lock` unless other locks are present.
+      codelingo/review:
+        comment: There are no other locks besides '{{lockname}}' here. Rename '{{lockname}}' to 'lock'.
+    query: |
+      import codelingo/ast/go
+      
+      go.file(depth = any):
```
Comment (rstorr) [inline tenets/codelingo/k8/well-named-lock/codelingo.yaml]: Remove unnecessary go.file.

File: tenets/codelingo/k8/flags-have-underscores/codelingo.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      codelingo/docs:
+        title: Flags Have Underscores
+        body: |
+          Command-line flags should use dashes, not underscores. See "https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md"
+      codelingo/review:
+        comment: Flag `{{flagname}}` contains an underscore. Use a dash instead.
+    query: |
+      import codelingo/ast/go
+      
+      go.file(depth = any):
```
Comment (rstorr) [inline tenets/codelingo/k8/flags-have-underscores/codelingo.yaml]: Remove unnecessary go.file.

File: tenets/codelingo/k8/jsonapi-compulsory-fields/codelingo.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    flows:
+      codelingo/docs:
+        title: JSON API Compulsory Fields
+        body: |
+      codelingo/review:
+        comment: This is a JSON object that is returned by an API but it does not have both a 'kind' and 'apiVersion' field.
+    query: |
+      import codelingo/ast/go
+            
+      go.file(depth = any):
```
Comment (rstorr) [inline tenets/codelingo/k8/jsonapi-compulsory-fields/codelingo.yaml]: Remove unnecessary go.file.

File: tenets/codelingo/k8/misused-nil-interface/codelingo.yaml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      codelingo/docs:
+        title: Misused Nil Interface
+        body: |
+          Nil interface is not the same as having a nil pointer in the interface.
+      codelingo/review:
+        comment: The variable being tested for nil, `{{varname}}` is a pointer to a pointer to nil.
+    query: |
+      import codelingo/ast/go
+      
+      go.file(depth = any):
```
Comment (rstorr) [inline tenets/codelingo/k8/misused-nil-interface/codelingo.yaml]: Remove unnecessary go.file.

File: tenets/codelingo/k8/well-named-locks/codelingo.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          Locks should be called `lock` and should never be embedded 
+          (always `lock sync.Mutex`). When multiple locks are present, 
+          give each lock a distinct name following Go conventions 
+          - `stateLock`, `mapLock` etc.
+      codelingo/review:
+        comment: Multiple locks are present, give each lock a distinct name following Go conventions - `stateLock`, `mapLock` etc.
+    query: |
+      import codelingo/ast/go
+      
+      go.file(depth = any):
```
Comment (rstorr) [inline tenets/codelingo/k8/well-named-locks/codelingo.yaml]: Remove unnecessary go.file.

File: tenets/codelingo/k8/use-informer-not-list/codelingo.yaml (hunk 1 above)
Comment (mullikine) [reply tenets/codelingo/k8/use-informer-not-list/codelingo.yaml]: Sorry, this tenet was not supposed to make it into the PR

File: tenets/codelingo/k8/new-package-requires-test/codelingo.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      import codelingo/ast/go
+
+      go.dir(depth = any):
+        filename as packageDirName
+        isNotTestPath(packageDirName)
+        go.file:
+          filename as packageFileName
+          @review comment
+          go.ident:
+            # This matches multiple things. Possibly because I am using go.dir.
```
[the remark this answers is not in the document]
Comment (mullikine) [reply tenets/codelingo/k8/new-package-requires-test/codelingo.yaml]: Last time I tested this tenet there were some issues around too many matches. This is one tenet that will have to be tested again. I will remove the comment.

**Pull request merged**